pkg/chunk/gcp/bigtable_index_client.go

```diff
@@ -49,16 +51,19 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&cfg.Instance, "bigtable.instance", "", "Bigtable instance ID.")
 	f.BoolVar(&cfg.TableCacheEnabled, "bigtable.table-cache.enabled", true, "If enabled, once a tables info is fetched, it is cached.")
 	f.DurationVar(&cfg.TableCacheExpiration, "bigtable.table-cache.expiration", 30*time.Minute, "Duration to cache tables before checking again.")
-
+	f.Float64Var(&cfg.WriteLimit, "bigtable.throttle-limit", 10.0, "BigTable rate cap to back off when throttled.")
```

Comment:
I would suggest to:
1. Call it `bigtable.write-rate-limit`
2. Defaults it to `0` (disabled)
3. Improve the flag description explaining it's the max number of keys written per second to bigtable, once this rate is exceeded writes are throttled and that 0 disables it

Reply:
@pracucci this is exactly what I was looking for, some pointers, will make changes and test it again.